src/fileio/impls/netcdf4/ncxx4.cxx

```diff
@@ -352,6 +352,97 @@ bool Ncxx4::addVarInt(const string &name, bool repeat) {
   return true;
 }
 
+bool Ncxx4::addVarIntVec(const string &name, bool repeat, size_t size) {
+  if(!is_valid())
+    return false;
+
+  NcVar var = dataFile->getVar(name);
+  const auto vec_dim_name = "vec" + std::to_string(size);
+  auto vec_dim = dataFile->getDim(vec_dim_name);
+  if (vec_dim.isNull()) {
+    vec_dim = dataFile->addDim(vec_dim_name, size);
+  }
+  if(var.isNull()) {
+    // Variable not in file, so add it.
+    if (repeat) {
+      auto dims = getRecDimVec(2);
+      dims[1] = vec_dim;
+      var = dataFile->addVar(name, ncInt, dims);
+    } else {
+      auto dims = getDimVec(1);
+      dims[0] = vec_dim;
+      var = dataFile->addVar(name, ncInt, dims);
+    }
+
+    if(var.isNull()) {
+      output_error.write("ERROR: NetCDF could not add int '{:s}' to file '{:s}'\n", name, fname);
+      return false;
+    }
+  } else {
+    // Check the existing variable is consistent with what's being added
+    if (repeat) {
+      ASSERT0(var.getDimCount() == 2);
+      if (var.getDim(1).getSize() != size) {
+        throw BoutException("Found existing variable '{:s}' with size {}. Trying to add "
+                            "with size {}.", name, var.getDim(1).getSize(), size);
+      }
+    } else {
+      ASSERT0(var.getDimCount() == 1);
+      if (var.getDim(0).getSize() != size) {
+        throw BoutException("Found existing variable '{:s}' with size {}. Trying to add "
+                            "with size {}.", name, var.getDim(0).getSize(), size);
+      }
+    }
```

Comment:
I didn't bother adding these checks to the hdf5 implementation, because there wasn't a simple way to add them there. Plus I don't think hdf5 is used much (at all?) and is probably being removed for v5.